### PR TITLE
Feature: add BoundedInt trait in `core::integer` for min() and max()

### PR DESCRIPTION
**Pull request by Amxx**
When manipulating u256, it is sometime necessary to represent the max value, similarly to how solidity does `type(uint256).max`. For example, some implementation of ERC-20 do not decrease the allowance in transferFrom if the current allowance is set to this max value.

Having this maximum value accessible from the corelib would save devs the burden of re-declaring this in their files.

I'm open to many different names such as `uint256_max`, `max_uint256`. If possible, `uint256::max` would be nice.

Note that this is declared as a function and not as a constant because `Only literal constants are currently supported.`

- - -
This change is [Reviewable](https://reviewable.io/reviews/starkware-libs/cairo/2563)

### CODE REVIEW

File: corelib/src/test.cairo [hunk 1]
```diff
@@ -654,6 +654,18 @@ fn test_u256_from_felt252() {
...
+#[test]
+fn test_max_u256() {
+    let max_u128 = 0xffffffffffffffffffffffffffffffff_u128;
+    assert(max_u256() == as_u256(max_u128, max_u128), 'not max');
+}
+
+#[test]
+#[should_panic]
+fn test_max_u256_plus_1_overflow() {
+    max_u256() + 1.into;
```
Comment (milancermak) [inline corelib/src/test.cairo:666]:
```suggestion
    max_u256() + 1.into();
```

**Review by spapinistarkware — COMMENTED** [review]
Suggestion: define a trait instead, something like
```
trait BoundedInt<T> {
  fn min() -> T;
  fn max() -> T;
}
```
implement it for u256. Then you could use `BoundedInt::max()`, or `BoundedInt::<u256>::max()` if not inferred. You will probably be able to use `u256::max()` in the future, though, I don't think this is currently supported.

> *[Reviewable](https://reviewable.io/reviews/starkware-libs/cairo/2563#-:-NRSI3Oc6EbUrr-b2bSw:bd1opxm)* status: 0 of 2 files reviewed, 1 unresolved discussion (waiting on @Amxx and @milancermak)

**Review by spapinistarkware — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/cairo/2563)* status: 0 of 2 files reviewed, 2 unresolved discussions (waiting on @Amxx and @milancermak)

___
*[`corelib/src/integer.cairo` line 982 at r1](https://reviewable.io/reviews/starkware-libs/cairo/2563#-NRSIwVn0isIWaVlkVxU:-NRSIwVn0isIWaVlkVxV:bvt91tj) ([raw file](https://github.com/starkware-libs/cairo/blob/1ff74d01a4a5e544820affc0ba66adcf319686bc/corelib/src/integer.cairo#L982)):*
> ```Cairo
> 
> fn max_u256() -> u256 implicits(RangeCheck) nopanic {
>     u256 { low: 0xffffffffffffffffffffffffffffffff_u128, high: 0xffffffffffffffffffffffffffffffff_u128 }
> ```

Line too long. Is your formatter configured correctly? WIth max line length of 100?

**Review by Amxx — COMMENTED** [review]
I refactored my PR to use this trait. I implemented it for u8, u16, u32, u64, u128 and u256

> *[Reviewable](https://reviewable.io/reviews/starkware-libs/cairo/2563#-:-NRYjGAO8g-wrDYiUIss:b-4g6bhl)* status: 0 of 2 files reviewed, 2 unresolved discussions (waiting on @milancermak and @spapinistarkware)

**Review by spapinistarkware — CHANGES_REQUESTED** [review]
> Reviewed 1 of 2 files at r2, 1 of 1 files at r3.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/cairo/2563)* status: all files reviewed (commit messages unreviewed), 3 unresolved discussions (waiting on @Amxx)

___
*[`corelib/src/integer.cairo` line 987 at r3](https://reviewable.io/reviews/starkware-libs/cairo/2563#-NRYohG89y6ifrnbX5a_:-NRYohG91jH6mKFnG8e3:bctudko) ([raw file](https://github.com/starkware-libs/cairo/blob/2e2868758953246e0a9e3440cebd7c2003bb94e9/corelib/src/integer.cairo#L987)):*
> ```Cairo
> }
> 
> impl BoundedU8 of BoundedInt::<u8> {
> ```

Please add `#[inline(always)]` above all of the impl functions. Also remove the RangeCheck.

**Review by spapinistarkware — APPROVED** [review]
:lgtm:

> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/cairo/2563#-:-NRar8Yp4pkBvSZOcrOc:bnfp4nl)* status: all files reviewed, 1 unresolved discussion (waiting on @Amxx)

**Review by Amxx — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/cairo/2563)* status: 0 of 3 files reviewed, 1 unresolved discussion (waiting on @milancermak and @spapinistarkware)

___
*[`corelib/src/integer.cairo` line 987 at r3](https://reviewable.io/reviews/starkware-libs/cairo/2563#-NRYohG89y6ifrnbX5a_:-NRbEfbd4QGwWRZw7vMa:bb7f2mf) ([raw file](https://github.com/starkware-libs/cairo/blob/2e2868758953246e0a9e3440cebd7c2003bb94e9/corelib/src/integer.cairo#L987)):*
<details><summary><i>Previously, spapinistarkware (Shahar Papini) wrote…</i></summary><blockquote>

Please add `#[inline(always)]` above all of the impl functions. Also remove the RangeCheck.
</blockquote></details>

done

**Review by Amxx — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/cairo/2563)* status: 0 of 3 files reviewed, 1 unresolved discussion (waiting on @milancermak and @spapinistarkware)

___
*[`corelib/src/test.cairo` line 666 at r1](https://reviewable.io/reviews/starkware-libs/cairo/2563#-NR-u61nMsIJPQyn2QLQ-r1-666:-NRbEl-f1diQPgny5gdW:b-896fix) ([raw file](https://github.com/starkware-libs/cairo/blob/1ff74d01a4a5e544820affc0ba66adcf319686bc/corelib/src/test.cairo#L666)):*
<details><summary><i>Previously, milancermak (Milan Cermak) wrote…</i></summary><blockquote>

```suggestion
    max_u256() + 1.into();
```
</blockquote></details>

Done.

**Review by spapinistarkware — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/cairo/2563)* status: all files reviewed, 1 unresolved discussion (waiting on @milancermak)

**Pull request merged**